includes/class-sensei-usage-tracking-data.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Usage tracking data
+ **/
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Supplies the usage tracking data for logging.
+ *
+ * @package Usage Tracking
+ * @since 1.9.20
+ */
+class Sensei_Usage_Tracking_Data {
+	/**
+	 * Get the usage tracking data to send.
+	 *
+	 * @since 1.9.20
+	 *
+	 * @return array Usage data.
+	 **/
+	public function get_usage_data() {
+		return array(
+			'courses' => wp_count_posts( 'course' )->publish,
+			'learners' => $this->get_learner_count(),
+			'lessons' => wp_count_posts( 'lesson' )->publish,
+			'messages' => wp_count_posts( 'sensei_message' )->publish,
+			'questions' => wp_count_posts( 'question' )->publish,
+			'teachers' => $this->get_teacher_count(),
+		);
+	}
+
+	/**
+	 * Get the number of teachers.
+	 *
+	 * @since 1.9.20
+	 *
+	 * @return int Number of teachers.
+	 **/
+	private function get_teacher_count() {
+		$teacher_query = new WP_User_Query( array( 'role' => 'teacher' ) );
+
+		return $teacher_query->total_users;
+	}
+
+	/**
+	 * Get the total number of learners enrolled in at least one course.
+	 *
+	 * @since 1.9.20
+	 *
+	 * @return int Number of learners.
+	 **/
+	private function get_learner_count() {
+		$learner_count = 0;
+		$args['fields'] = array( 'ID' );
+		$user_query = new WP_User_Query( $args );
+		$learners = $user_query->get_results();
+
+		foreach( $learners as $learner ) {
+			$course_args = array(
```

Comment:
@pgk Would you mind verifying this logic? Is this the proper way to check if a user is currently enrolled in or has already completed any course?

Reply:
I would use this: https://github.com/Automattic/sensei/blob/master/includes/class-sensei-utils.php#L1595

Reply:
Oh you wrote *any* course? Sorry, misread this. Yeah, LGTM